src/game/board/boardSituation.js

```diff
@@ -1,14 +1,31 @@
-export default class BoardSituation {
-    constructor() {
+export default class BoardSituation {
+        constructor(firstBoardPosition) {
+            if(newBoardPosition instanceof  BoardPosition){
+                this.boardPosition=firstBoardPosition;
+                this.boardSituation=[];
+            }
+            else {
+                 window.alert('Incorrect data given');
+            }
+        }
+        get position() {
+            // TODO: this function should return current position on the board as an 8x8 array of ChessPiece objects
+            return boardPosition;
+        }
+        get previousSituation() {
+            // TODO: this function should return previous situation or null if there's none (in case of the first position)
+            return boardSituation;
+        }
+       
+        setNewBoardSituation(newBoardPosition)
+        {
+            if(newBoardPosition instanceof  BoardPosition){
+            this.boardSituation.unshift(this.boardPosition);
+            this.boardPosition=newBoardPosition;
+            }
+            else {
+                window.alert('Incorrect data given');
+            }
+        }
```

Comment:
Ta metoda nie powinna istnieć. Jeśli miałaby istnieć metoda tworząca nową sytuację ze starej, to powinna się nazywać inaczej i robić co innego (widzę tu operacje tablicowe na czymś, co nigdy nie powinno być tablicą).

Tworzenie nowej sytuacji jako funkcja starej (można by to rozważyć jako alternatywę dla opcji konstruktora, o której pisałem wcześniej) nie powinno zmieniać obiektu starej sytuacji - powinno zwrócić nowy obiekt klasy BoardSituation (dla którego previousSituation zwracałoby obiekt, na którym funkcja tworząca nową sytuację została wywołana).